tests/ui/ref_as_ptr.rs

```diff
@@ -58,27 +64,27 @@ fn _msrv_1_75() {
     let mut_val = &mut 42_i32;
 
     // `std::ptr::from_{ref, mut}` was stabilized in 1.76. Do not lint this
-    let _ = val as *const i32;
-    let _ = mut_val as *mut i32;
+    f(val as *const i32);
+    f(mut_val as *mut i32);
 }
 
 #[clippy::msrv = "1.76"]
 fn _msrv_1_76() {
     let val = &42_i32;
     let mut_val = &mut 42_i32;
 
-    let _ = val as *const i32;
-    let _ = mut_val as *mut i32;
+    f(val as *const i32);
+    f(mut_val as *mut i32);
 }
 
 fn foo(val: &[u8]) {
-    let _ = val as *const _;
```

Comment:
Similarly, `let _ = val as *const _` should absolutely trigger the lint, as there is no lifetime extension.